plugins/abnormal_security/icon_abnormal_security/util/api.py

```diff
@@ -64,16 +72,22 @@ def send_request(self, method: str, path: str, params: dict = None, payload: dic
         except requests.exceptions.HTTPError as e:
             raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
 
+    def generate_filter_params(self, from_date: str = None, to_date: str = None) -> dict:
+        params = {}
+        if from_date or to_date:
+            params = {"filter": "receivedTime"}
+            if from_date:
+                params["filter"] = params["filter"] + f" gte {self.parse_date(from_date)}"
+            if to_date:
+                params["filter"] = params["filter"] + f" lte {self.parse_date(to_date)}"
+        return params
+
     @staticmethod
-    def validate_iso8601(dt_string):
+    def parse_date(date: str) -> str:
         try:
-            datetime.fromisoformat(dt_string)
-        except ValueError:
-            try:
-                datetime.fromisoformat(dt_string.replace("Z", "+00:00"))
-            except ValueError:
-                raise PluginException(
-                    cause=f"Date: {dt_string} is not a valid ISO8601 date.",
-                    assistance="Please update the date to match ISO8601 format (YYYY-MM-DDTHH:MM:SSZ).",
-                )
-        return True
+            return dateparser.parse(date).isoformat()
+        except AttributeError:
```

Comment:
```suggestion
        except Exception:
```
Here we would want to catch all exceptions. We looked into the source of `dateparser.parse()` and it may return multiple types of exceptions such as `ValueError`, `TypeError`, or `SettingValidationError`.